### PR TITLE
V14: Add required dependency to management api

### PR DESCRIPTION
**Pull request by Zeegaan**
Fixes https://github.com/umbraco/Umbraco-CMS/issues/16516

# Notes
- Adds registration to management API aswell
- Adds check to see if we have already registered, to not register again

# How to test
- Remove `.AddDeliveryApi()` from `Program.cs`
- Program should be able to run

### CODE REVIEW

**Review by andr317c — APPROVED** [review]
LGTM!

File: src/Umbraco.Cms.Api.Common/DependencyInjection/UmbracoBuilderApiExtensions.cs [hunk 1]
```diff
@@ -11,6 +11,11 @@ public static class UmbracoBuilderApiExtensions
 {
     public static IUmbracoBuilder AddUmbracoApiOpenApiUI(this IUmbracoBuilder builder)
     {
+        if(builder.Services.Any(x => x.ImplementationType == typeof(OperationIdSelector)))
```
Comment (Nuklon) [inline src/Umbraco.Cms.Api.Common/DependencyInjection/UmbracoBuilderApiExtensions.cs:14]:
```suggestion
        if (builder.Services.Any(x => x.ImplementationType == typeof(OperationIdSelector)))
```

**Comment by AnnaTornroth-NLTG** [comment]
Hi @iOvergaard and @Zeegaan, 
This issue still remains in version 17.1.0, can you please have a look?

**Comment by AndyButland** [comment]
I'm not seeing any problem running 17.1 with the delivery API disabled @AnnaTornroth-NLTG - can you share some more details please?

To test I've created a new 17.1 Umbraco website and amended `Program.cs` as follows:

```
builder.CreateUmbracoBuilder()
    .AddBackOffice()
    .AddWebsite()
    //.AddDeliveryApi()
    .AddComposers()
    .Build();
```

**Comment by AnnaTornroth-NLTG** [comment]
Hi,
I apologize for the mixed information, @AndyButland.

In the API you have to do this:
image


Is this something you can fix?

**Comment by AndyButland** [comment]
That's a different issue then (this one was about the delivery AP).  It's worth raising as it should be something we can support, but please create a new issue for this, and provide details of what problems you encounter when you try to use Umbraco without the backoffice registered.

**Comment by AnnaTornroth-NLTG** [comment]
Ok, created https://github.com/umbraco/Umbraco-CMS/issues/21622 :)

**Pull request merged**